src/Symfony/Contracts/Translation/TranslatableInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\Translation;
+
+/**
+ * @author Nate Wiebe <nate@northern.co>
+ */
+interface TranslatableInterface
+{
+    /**
+     * @return string The message id (may also be an object that can be cast to string)
+     */
+    public function getMessage(): string;
+
+    /**
+     * @return array An array of parameters for the message
+     */
+    public function getParameters(): array;
```

Comment:
i tend to believe domain/locale are as useful.

Reply:
Right, the domain should be here, but not sure about the locale as we may want to translate with a different language depending on the situation. That decision should likely be separate from where we're creating the translatable object. Thoughts?

Reply:
right, locale could be nullable. I tend to believe a "Translatable" may represent a message ID in a specific locale, but by default uses the "current locale" aka null.

Reply:
Okay, so the only other thing then would be what (if anything) should be allowed to be overwritten/merged when doing the actual translation. I would assume that the message id and domain shouldn't. I'm going back and forth in my head with the parameters, though I'm leaning towards allowing them to merge. If we don't add that ability, people might just do workarounds in the templates which I think might be worse. Locale I think should be allowed to be overwritten in case you want to display the message in the view with the user's locale but log or email to an admin/dev in a different locale.

Reply:
IMHO if the domain/locale is passed additionally, it overwrites the Translatable. Yet, parameters could be merged, as it effectively overwrites parameters keys anyway.

Reply:
Sounds good, I'll make these changes.

Reply:
Thinking about this more, I feel like it might be mixing responsibilities a bit adding in locale. I feel like the Translatable class should be responsible for *what's* being said and the translator for *how* it's said. So the Translatable should contain the message (message id, context data, domain), but the locale is part of how it's presented and where you create a translatable likely shouldn't know what the locale is going to be.

The only exception to this rule currently is the parameters, but potentially there is some instances where you might want to manipulate the data a bit based on where and how it's going to be said, but that's probably fine because the parameters could be contextual.

Reply:
i think you're right. A single translatable (without locale) can be translated multiple times in different locales, that sounds more logical to do than having a translatable per locale in general, i agree.

It's also one less nullable, which is nice :)